llvm/lib/Target/AArch64/AArch64Subtarget.cpp

```diff
@@ -491,13 +491,15 @@ bool AArch64Subtarget::isStreamingCompatible() const {
 }
 
 bool AArch64Subtarget::isNeonAvailable() const {
-  return hasNEON() && !isStreaming() && !isStreamingCompatible();
+  if (hasSMEFA64())
+    return true;
+  return (hasNEON() && !isStreaming() && !isStreamingCompatible());
 }
 
-bool AArch64Subtarget::isSVEAvailable() const{
-  // FIXME: Also return false if FEAT_FA64 is set, but we can't do this yet
-  // as we don't yet support the feature in LLVM.
-  return hasSVE() && !isStreaming() && !isStreamingCompatible();
+bool AArch64Subtarget::isSVEAvailable() const {
+  if (hasSMEFA64())
```

Comment:
Like above, this also needs to check for `hasSVE()`

Reply:
Done